apps/src/templates/teacherNavigation/lessonMaterials/LessonMaterialsContainer.tsx

```diff
@@ -96,15 +74,48 @@ interface LessonMaterialsContainerProps {
 const LessonMaterialsContainer: React.FC<LessonMaterialsContainerProps> = ({
   showNoCurriculumAssigned,
 }) => {
-  const loadedData = useLoaderData() as LessonMaterialsData | null;
+  const [lessonMaterials, setLessonMaterials] =
+    useState<LessonMaterialsData | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const selectedSection = useAppSelector(selectedSectionSelector);
+
+  React.useEffect(() => {
+    const fetchLessonMaterials = async () => {
+      const state = getStore().getState().teacherSections;
+      const selectedSectionId = state.selectedSectionId;
+      const sectionData = state.sections[selectedSectionId];
+
+      if (!selectedSectionId || !sectionData.unitId) {
+        setLessonMaterials(null);
+        setIsLoading(false);
+        return;
+      }
+
+      setIsLoading(true);
+
+      const data = await lessonMaterialsCachedLoader(sectionData.unitId);
+      setLessonMaterials(data);
+      setIsLoading(false);
+
+      if (data?.unitName) {
+        analyticsReporter.sendEvent(EVENTS.VIEW_LESSON_MATERIALS, {
+          unitName: data.unitName,
+        });
+      }
+    };
```

Comment:
```suggestion
      const data = await lessonMaterialsCachedLoader(sectionData.unitId)
      .then(data => {
        setLessonMaterials(data);
        setIsLoading(false);
  
        if (data?.unitName) {
          analyticsReporter.sendEvent(EVENTS.VIEW_LESSON_MATERIALS, {
            unitName: data.unitName,
          });
        }
      }
    };
```
If we structure it using a `.then()` instead of `await` then we don't need to define `fetchLessonMaterials` and then call it later. This is also that pattern that most other places we have recently been fetching with use.